app/controllers/admin/frameworks_controller.rb

```diff
@@ -2,4 +2,34 @@ class Admin::FrameworksController < AdminController
   def index
     @frameworks = Framework.all
   end
+
+  def new
+    @framework = Framework.new
+  end
+
+  def show
+    @framework = Framework.find(params[:id])
+  end
+
+  def create
+    definition_source = framework_params[:definition_source]
+    framework_def = Framework::Definition::Language.generate_framework_definition(definition_source)
+
+    @framework =  Framework.new(
+      name:              framework_def.name,
+      short_name:        framework_def.framework_short_name,
+      definition_source: definition_source
+    )
+
+    if @framework.save
+      flash[:success] = 'Definition saved successfully'
+      redirect_to admin_framework_path(@framework)
+    end
+  end
+
+  private
+
+  def framework_params
```

Comment:
Seeing as we're not mass assigning attributes, I don't think we need to use strong params here.

Reply:
There isn't a downside to having it; we *could* replace
```ruby
    params.require(:framework).permit(:definition_source)
```

with

```ruby
    params.require[:framework][:definition_source]
end
```

... but that would just downgrade the quality of error message were `:framework` not to be supplied?

I suppose I quite like having it for the sake of consistency and safety; it's the sort of thing that tends to get copied by newer devs, its presence generally makes controllers read more cleanly, and it comes at very little cost, so even in the absence of mass assignment – why not?